lms/static/js/dashboard/legacy.js

```diff
@@ -153,27 +224,27 @@
          });
 
          $('.action-email-settings').each(function(index) {
-             $(this).attr('id', 'email-settings-' + index);
             // a bit of a hack, but gets the unique selector for the modal trigger
              var trigger = '#' + $(this).attr('id');
              accessibleModal(
                 trigger,
                 '#email-settings-modal .close-modal',
                 '#email-settings-modal',
                 '#dashboard-main'
-            );
+             );
+             $(this).attr('id', 'email-settings-' + index);
```

Comment:
Becase of run_quality was failing and asking to move all defined var to top of the function.